### PR TITLE
refactor(operators): convert operators to typescript

### PR DESCRIPTION
**Pull request by fncolon**
### Pull Request Checklist

_Please make sure to review and check all of these items:_

- [ ] Have you added new tests to prevent regressions?
- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

To avoid name clashes with your global symbol keys and other (library code) global symbols, it might be a good idea to prefix your symbols

### Todos

- [x] Migrate operators.js to typescript file

### CODE REVIEW

**Comment by fncolon** [comment]
Since papb no longer active maintaining directly in the repository, so I assign @sdepold to review this.

**Comment by WikiRik** [comment]
Could you maybe migrate this file to TypeScript as well? Otherwise we will have to change this file twice in a short time

**Comment by fncolon** [comment]
> Could you maybe migrate this file to TypeScript as well? Otherwise we will have to change this file twice in a short time

Okay, will work on it, drafted

File: .eslintrc.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
         "avoidEscape": true
       }
     ],
-    "linebreak-style": "error",
+    "linebreak-style": ["error", "windows"],
```
Comment (WikiRik) [inline .eslintrc.json:72]: Why would we want to do CRLF line endings?

File: lib/operators.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+
+'use strict';
```
Comment (WikiRik) [inline lib/operators.ts:2]: This is not needed for TypeScript right?

File: lib/operators.js [hunk 3]
```diff
@@ -46,46 +46,46 @@
  * @property join
```
Comment (WikiRik) [inline lib/operators.js:46]: When replacing this file with a TypeScript file, we don't need this file anymore right?

File: .eslintrc.json (hunk 1 above)
Comment (fncolon) [reply .eslintrc.json:72]: hello i didn't done it yet, please wait undrafted

Comment (WikiRik) [reply .eslintrc.json:72]: Sorry, then I'll let you continue and wait until it's ready for review. Thanks for working on it!

File: types/lib/operators.d.ts [hunk 4]
```diff
@@ -481,4 +481,4 @@ declare const Op: {
   readonly values: unique symbol;
 };
 
-export = Op;
+export = Op;
```
Comment (WikiRik) [inline types/lib/operators.d.ts:484]: This change is not wanted

File: .eslintrc.json [hunk 5]
```diff
@@ -69,7 +69,7 @@
         "avoidEscape": true
       }
     ],
-    "linebreak-style": "error",
+    "linebreak-style": ["error"],
```
Comment (WikiRik) [inline .eslintrc.json:72]: Is this change needed?

**Review by WikiRik — COMMENTED** [review]
TypeScript changes look okay, but I'll leave it open for people with more experience. I do have some remarks about the other changes you made

File: types/lib/operators.d.ts (hunk 4 above)
Comment (fncolon) [reply types/lib/operators.d.ts:484]: look this changed the line backj to LF. which is follow our eslint rules.

Comment (WikiRik) [reply types/lib/operators.d.ts:484]: GitHub shows me that there is no empty line at the end of the file, which is preferred

**Comment by sdepold** [comment]
I added some last bits to the PR
